### ISSUE TITLE
GzipHandler seems to be capable of adding duplicate Vary headers

### ISSUE THREAD

**Issue by joakime** [eclipse/jetty.project#383]
Under some conditions, the `Vary` header seems to be added twice to the response header.

**Comment by joakime** [comment]
The Vary header is added by the GzipHttpOutputInterceptor

See: https://github.com/eclipse/jetty.project/blob/master/jetty-server/src/main/java/org/eclipse/jetty/server/handler/gzip/GzipHttpOutputInterceptor.java#L190-L191

**Comment by sbordet** [comment]
Do you have more information about this issue ? "Some conditions" is not really reproducible :)

**Comment by kviphong** [comment]
This is a clean version of what is going on in my code:

```
public class SpecialHandler extends GzipHandler {
    @override
    public void handle(String target, Request baseRequest, HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
        super.handle(target, baseRequest, request, response);
    }
}
```

Set this SpecialHandler for my WebApp. Then, send request, doesn't matter what its Accept-encoding is. The response will have 2 Vary headers.

I haven't tried this out myself on a vanila Jetty installation, so, not 100% sure it'll be able to repro the issue. What I listed is my best knowledge on how I understand the issue. Hope this helps.




Thanks,
Tyler

**Comment by sbordet** [comment]
@kviphong I don't understand. Your `SpecialHandler` is an empty subclass of `GzipHandler`, so if you get double headers, you should get them with a vanilla `GzipHandler`.

What is your complete server code ? Perhaps you add the `GzipHandler` twice.

Do you have server DEBUG logs ?

Can you take a network trace of the request/response that contains the 2 `Vary` headers ?

**Comment by joakime** [comment]
This seems to have been fixed as part of the effort in Issue #406
